web/src/js/utils.js

```diff
@@ -105,3 +105,17 @@ fetchApi.put = (url, json, options) => fetchApi(
         ...options
     }
 )
+
+
+export function getDiff(obj1, obj2) {
+    let result = {...obj2};
+    _.forIn(obj1, (value, key) => {
+    	  if(_.isEqual(obj2[key], obj1[key]))
+            result[key] = undefined;
+        else if(typeof Array.isArray(obj2[key]) && Array.isArray(obj2[key]))
+            result[key] = {...obj2[key]};
```

Comment:
My understanding is that we would like to send the whole headers array on update, not just parts of it. So if the headers are not equal, we should just send the whole array. Otherwise, updating on the server-side would be relatively tricky.